crates/core/src/config/custom.rs

```diff
@@ -1,8 +1,15 @@
-use std::sync::Arc;
+use std::{collections::HashMap, sync::Arc};
 
 use crate::context::CompilationContext;
 
+use super::{
+  config_regex::ConfigRegex,
+  external::{ExternalConfig, ExternalObject},
+  Config,
+};
+
 const CUSTOM_CONFIG_RUNTIME_ISOLATE: &str = "runtime.isolate";
+pub const CUSTOM_CONFIG_EXTERNAL_RECORD: &str = "RECORD_EXTERNAL";
```

Comment:
Using `external.record` would be better to normalize the name convention and make it more readable